### PR TITLE
X 3scale debug headers

### PR DESCRIPTION
**Pull request by palmieric**
Added X-3scale-debug-headers also when a mapping rule is not matched

### CODE REVIEW

**Comment by eloycoto** [comment]
Code LGTM, If you can add a test on `t/apicast-mapping-rules.t` it'll be great!

**Comment by palmieric** [comment]
> Code LGTM, If you can add a test on `t/apicast-mapping-rules.t` it'll be great!

Sure! I'll do it.

File: t/apicast-mapping-rules.t [hunk 1]
```diff
@@ -323,3 +323,53 @@ GET /foo%20/bar/?user_key=foo
...
+--- request
+POST /bar
+--- more_headers
+X-3scale-Debug: my-token
+--- response_body
+api response
+--- error_code: 404
+--- response_headers
+X-3scale-matched-rules: /foo?bar=baz
+X-3scale-usage: usage%5Bbar%5D=7
```
Comment (eloycoto) [inline t/apicast-mapping-rules.t:375]: This is not true no? response should not have this value no?

**Comment by eguzki** [comment]
Is there any progress on this PR?

**Comment by palmieric** [comment]
@eguzki @eloycoto I just fixed the test for the mapping rules, now it should be closed/merged

**Comment by kevprice83** [comment]
Thanks @palmieric would you be able to resolve the conflicts?

**Comment by palmieric** [comment]
> 

@kevprice83 @eguzki @eloycoto fixed the merge conflicts too

File: t/apicast-mapping-rules.t [hunk 2]
```diff
@@ -323,3 +323,47 @@ GET /foo%20/bar/?user_key=foo
...
+
+  location /api/ {
+    echo "api response";
+  }
+
+  location /transactions/authrep.xml {
+    content_by_lua_block { ngx.exit(200) }
+  }
+--- request
+POST /bar
```
Comment (kevprice83) [inline t/apicast-mapping-rules.t:366]: @palmieric this test is failing because there are no credentials provided in the request

**Review by kevprice83 — CHANGES_REQUESTED** [review]
Tests are currently failing and Test case shuold be numbered 7 NOT 6 :)

File: t/apicast-mapping-rules.t (hunk 2 above)
Comment (palmieric) [reply t/apicast-mapping-rules.t:366]: I didn't really test it.
Will fix it soon.

**Comment by eguzki** [comment]
any update? Additionally, conflicts must be resolved.

**Comment by eguzki** [comment]
Closing it 

Reopen if needed

**Pull request closed without merging**